chart/values.yaml

```diff
@@ -136,3 +136,17 @@ customLogos:
   # storageClass: "-"
   accessMode: ReadWriteOnce
   size: 1Gi
+
+# Rancher post-delete hook
+postDelete:
+  enabled: true
+  name: rancher-post-delete
+  image:
+    repository: rancher/shell
+    tag: v0.1.6
```

Comment:
This will not get pulled into images.txt if the primary shell gets updated. That is from here: 
https://github.com/rancher/rancher/blob/master/pkg/settings/setting.go#L94
https://github.com/rancher/rancher/blob/master/pkg/image/resolve.go#L248

Can we use the agent ? I can't think of a way to get the shell into the chart consistently. Unless we hardcode it into the requirements struct there but that feels bad.

Reply:
Actually now that I think about it, I'm not sure the agent image exists on the local cluster in a default (non-airgap/public) setup, do you know ? It would suck to have to pull an image to start a post-delete hook. Its likely that the shell image would have been pulled into local, and if not its 1/4th the size of the agent. Is there anything else we can guarantee is on the local cluster ?

Reply:
This image `rancher/shell` should already be in the local cluster because the `helm-operation` pod uses it to install Helm charts in the local cluster. 
```
> k -n cattle-system describe pod helm-operation-jp4m2 | grep -i image
    Image:         rancher/shell:v0.1.6
    Image ID:      docker.io/rancher/shell@sha256:bc884e3b37a737a7e411766c42705fbe477d6219c72ec781efd8438d8fb267d0
    Image:         rancher/shell:v0.1.6
    Image ID:      docker.io/rancher/shell@sha256:bc884e3b37a737a7e411766c42705fbe477d6219c72ec781efd8438d8fb267d0
  Normal  Pulled     6m58s  kubelet            Container image "rancher/shell:v0.1.6" already present on machine
  Normal  Pulled     6m58s  kubelet            Container image "rancher/shell:v0.1.6" already present on machine
```

Still, the problem is that how we can get the image dynamically instead of hardcoded in the value.yaml file.

Reply:
> This will not get pulled into images.txt if the primary shell gets updated. That is from here:
> [https://github.com/rancher/rancher/blob/master/pkg/settings/setting.go#L94](https://github.com/rancher/rancher/blob/master/pkg/settings/setting.go?rgh-link-date=2021-03-16T02%3A36%3A58Z#L94)
> [https://github.com/rancher/rancher/blob/master/pkg/image/resolve.go#L248](https://github.com/rancher/rancher/blob/master/pkg/image/resolve.go?rgh-link-date=2021-03-16T02%3A36%3A58Z#L248)
> 
> Can we use the agent? I can't think of a way to get the shell into the chart consistently. Unless we hardcode it into the requirements struct there but that feels bad.


I have 2 possible solutions so far:
1. add some logic to the file [scripts/chart/build](https://github.com/rancher/rancher/blob/master/scripts/chart/build) to get the image and tag from the following line in [pkg/settings/setting.go](https://github.com/rancher/rancher/blob/2dca0c67b48331c23c05c025208bacfc258d45bd/pkg/settings/setting.go#L94)
```
	ShellImage                        = NewSetting("shell-image", "rancher/shell:v0.1.6")
```

2. add a comment to the above line in `pkg/settings/setting.go`:  we need to update the `postDelete.image` section in the file `/chart/values.yaml` for any change on this line.

Reply:
1. could work but it needs to load the entire rancher codebase for it to work, so that would take good amount of time.
2. If someone forgets then airgap won't work for this, much safer to add to requirements struct and also a comment there. That seems like best solution for now.